sql/core/src/main/scala/org/apache/spark/sql/execution/Generate.scala

```diff
@@ -75,9 +78,15 @@ case class Generate(
             outputRows.map(or => joinProjection(joinedRow(row, or)))
           }
         }
+      }.mapPartitions { iter =>
+        val nullOriginalRow = Row(Seq.fill(generator.output.size)(Literal(null)): _*)
```

Comment:
My mistake, I see you're building the `Row`. But here we should use `child.output.size` rather than `generator.output.size`. Also we can use `Row.fromSeq` instead of `Row.appy(values: Any*)` for clarity:

```scala
val nullOriginalRow = Row.fromSeq(Seq.fill(child.output.size)(Literal(null))
```